lib/logstash/outputs/elasticsearch.rb

```diff
@@ -531,4 +540,15 @@ def check_action_validity
 
     raise LogStash::ConfigurationError, "Action '#{@action}' is invalid! Pick one of #{valid_actions} or use a sprintf style statement"
   end
+
+  def check_dlq_custom_codes
+    intersection = dlq_custom_codes & DOC_DLQ_CODES
+    raise LogStash::ConfigurationError, "dlq_custom_codes contains error codes already used: #{intersection}" unless intersection.empty?
+
+    intersection = dlq_custom_codes & DOC_SUCCESS_CODES
+    raise LogStash::ConfigurationError, "dlq_custom_codes contains error codes already defined as success: #{intersection}" unless intersection.empty?
+
+    intersection = dlq_custom_codes & [DOC_CONFLICT_CODE]
+    raise LogStash::ConfigurationError, "dlq_custom_codes contains error codes already defined as conflict: #{intersection}" unless intersection.empty?
+  end
```

Comment:
besides success, conflict and dlq codes, there is a `failure_type_logging_whitelist`. Do you think it should be checked here?

Reply:
`failure_type_logging_whitelist` is used to log or not errors during the execution of the queries, while the method `#check_dlq_custom_codes` is used during registration to report configuration error. I think are different aspect, one concern about a configuration problem that should be logged or raised independently of `failure_type_logging_whitelist` which is a setting used to limit the number of logs.